spec/jobs/sufia_resque_queue_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+
+describe Sufia::Resque::Queue do
+  let(:subject) { Sufia::Resque::Queue.new "test_queue" }
+  let(:job) { double('job') }
+
+  context "with no retries" do
+    it "queues the job" do
+      expect(::Resque).to receive(:enqueue_to).once.and_return(true)
+      subject.push(job)
+    end
+  end
+
+  context 'when one run times out' do
+    before do
+      call_count = 0
+      allow(::Resque).to receive(:enqueue_to) do
+        call_count += 1
```

Comment:
I don't love this implementation, but I've tried to disentangle the code that makes a failure get raised from the code that checks to see that the job gets pushed.  The implementation before seemed to use `expect` as both a stub and a test. IOW, using `expect` asserts that the code below it (`subject.push(job)`) is expected to raise a timeout error -- why would `subject.push(job)` raise a timeout error on that line without prior `allow` stubs to create that error condition?

Not trying to split hairs here! Just trying to understand how your test was passing, and if it was doing what we thought it should do (because in the Sufia test suite we usually use `expect` for triggering tests and `allow` for setting up behaviors). Thoughts, @jcoyne ?  The original test here was:

```ruby
  expect(::Resque).to receive(:enqueue_to).and_raise(Redis::TimeoutError)
  expect(::Resque).to receive(:enqueue_to).and_return(true)
  subject.push(job)
```